api/envoy/config/core/v3/base.proto

```diff
@@ -339,12 +358,22 @@ message KeyValueAppend {
     OVERWRITE_IF_EXISTS = 3;
   }
 
-  // Key/value pair entry that this option to append or overwrite.
-  KeyValue entry = 1 [(validate.rules).message = {required: true}];
+  // Key/value pair entry that this option to append or overwrite. This field is deprecated
+  // and please use :ref:`content <envoy_v3_api_field_config.core.v3.KeyValueAppend.content>`
```

Comment:
Yes. This should be replaced by the `record` because we won't provide the `KeyValue`'s support. But to avoid a api break change (although this should never be used, but it wad added for a while), I can only deprecate and hide it, but cannot replace/change it directly.

Reply:
Good catch. This is a type error. And because the `[#not-implemented-hide:]`, the ci also didn't find this error.

Reply:
Fixed it.